Wrong apiVersion for "type": "Microsoft.Resources/deployments" in master.template.json

'Create' action generates 'master.deployment.json' according to .yml file. 

However during master.template.json deployment it fails with:

**"'code': 'InvalidTemplate', 'message': "Deployment template validation failed: 'The template resource 'globalServicePolicyTemplate' at line '19' and column '5' is invalid. 'ApiVersion' property is invalid. The supported versions are '2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,202"**

in master.deployment.json I can see there not supported for "Microsoft.Resources/deployments" apiVersion added:

"apiVersion": "2021-08-01",
      "type": "Microsoft.Resources/deployments",
      "name": "productsTemplate",

@adrianhall , i am also having same issue with beta version or the current main repo. It happens in dev ops pipeline when trying to validate generated apim files. Not sure whats the issue, tried the last release as well still have the same issue. Any help greatly appreciated .. I just want to extract complete apim download from one subscription and upload it to the next one, tried lots of version, sometime getting tags error or if tag error is gone then this validation error, its not isolated to just this global policy file, it has also come with other files namevalue.. i think since the latest code has tried to update the API version to 2020-08-01, this issue is introduced

`Checking if the following resource group exists: rg-test-300.
Resource group exists: true.
Creating deployment parameters.
The detected encoding for file 'apimExtract-prod\apim--prod-master.template.json' is 'utf-8'
(node:1780) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
The detected encoding for file '\apimExtract-prod\apim-prod-parameters.json' is 'utf-8'
Starting template validation.
Deployment name is apim-prod-master-20220411-225931-2bbb
There were errors in your deployment. Error code: InvalidTemplate.
##[error]Deployment template validation failed: 'The template resource 'globalServicePolicyTemplate' at line '1' and column '786' is invalid. 'ApiVersion' property is invalid. The supported versions are '2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'. Please see https://aka.ms/arm-template/#resources for usage details.'.
`
`{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "ApimServiceName": {
      "type": "string"
    },
    "PolicyXMLBaseUrl": {
      "type": "string"
    }
  },
  "resources": [
    {
      "properties": {
        "value": "[concat(parameters('PolicyXMLBaseUrl'), '/globalServicePolicy.xml')]",
        "format": "rawxml-link"
      },
      "name": "[concat(parameters('ApimServiceName'), '/policy')]",
      "type": "Microsoft.ApiManagement/service/policies",
      "apiVersion": "2019-01-01"
    }
  ]
}`

@DeagleGross 
Thanks for the link, i tried extracting using that link. Dont have any issue with extraction part. However, when i try to validate this in the pipeline i am still getting the same error 


`There were errors in your deployment. Error code: InvalidTemplate.
##[error]Deployment template validation failed: 'The template resource 'namedValuesTemplate' at line '1' and column '786' is invalid. 'ApiVersion' property is invalid. The supported versions are '2022-01-01,2021-04-01,2021-01-01,2020-10-01,2020-09-01,2020-08-01,2020-07-01,2020-06-01,2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'. Please see https://aka.ms/arm-template/#resources for usage details.'.
##[error]Check out the troubleshooting guide to see if your issue is addressed: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment?view=azure-devops#troubleshooting
##[error]Task failed while creating or updating the template deployment.
`

this is the output of apiversion in namevaluetemplate file
`{
      "apiVersion": "2021-08-01",
      "type": "Microsoft.ApiManagement/service/namedValues",
      "name": "[concat(parameters('apimServiceName'), '/auth-200')]",
      "properties": {
        "secret": false,
        "displayName": "auth-200",
        "value": "12345"
      }
    }`

I believe the issue is with the code which is outputting apiversion 2021-08-01

--- a/src/ArmTemplates/Common/Constants/GlobalConstants.cs
+++ b/src/ArmTemplates/Common/Constants/GlobalConstants.cs
@@ -12,7 +12,7 @@
         public const string ExtractDescription = "Extract an existing API Management instance";
 
         public const string ApiVersion = "2021-08-01";
-        public const string LinkedAPIVersion = "2021-08-01";
+        public const string ArmApiVersion = "2022-01-01";
         public const int NumOfRecords = 100;
 
         public const string azAccessToken = "account get-access-token --query \"accessToken\" --output json";

--- a/src/ArmTemplates/Creator/TemplateCreators/MasterTemplateCreator.cs
+++ b/src/ArmTemplates/Creator/TemplateCreators/MasterTemplateCreator.cs
@@ -227,7 +227,7 @@ namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Creator.Template
             {
                 Name = name,
                 Type = "Microsoft.Resources/deployments",
-                ApiVersion = GlobalConstants.LinkedAPIVersion,
+                ApiVersion = GlobalConstants.ArmApiVersion,
                 Properties = new MasterTemplateProperties()
                 {
                     mode = "Incremental",

--- a/src/ArmTemplates/Extractor/EntityExtractors/MasterTemplateExtractor.cs
+++ b/src/ArmTemplates/Extractor/EntityExtractors/MasterTemplateExtractor.cs
@@ -270,7 +270,7 @@ namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Entity
             {
                 Name = name,
                 Type = "Microsoft.Resources/deployments",
-                ApiVersion = GlobalConstants.LinkedAPIVersion,
+                ApiVersion = GlobalConstants.ArmApiVersion,
                 Properties = new MasterTemplateProperties()
                 {
                     mode = "Incremental",

